File: data/src/main/java/org/oppia/data/backends/gae/model/StateClassifier.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

public class StateClassifier {

  /*
   * Parent Hierarchy: ExplorationContainer -> StateClassifier
   */

  @Json(name = "algorithm_id")
  public String algorithmId;

  //TODO: Check this free form data
  @Json(name = "classifier_data")
  public String classifierData;

  @Json(name = "data_schema_version")
  public int dataSchemaVersion;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/StateClassifier.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class StateClassifier(

  @Json(name = "algorithm_id") val algorithmId: String?,
  @Json(name = "classifier_data") val classifierData: Any?,
  @Json(name = "data_schema_version") val dataSchemaVersion: Int?

)
